src/EnergyPlus/HeatBalanceIntRadExchange.cc

```diff
@@ -399,54 +426,92 @@ namespace HeatBalanceIntRadExchange {
 
                 // Calculate net long-wave radiation for opaque surfaces and incident
                 // long-wave radiation for windows.
-                if (construct.TypeIsWindow) {          // Window
-                    Real64 scriptF_acc(0.0);           // Local accumulator
-                    Real64 netLWRadToRecSurf_cor(0.0); // Correction
-                    Real64 IRfromParentZone_acc(0.0);  // Local accumulator
-                    for (size_type SendZoneSurfNum = 0; SendZoneSurfNum < s_zone_Surfaces; ++SendZoneSurfNum, ++lSR) {
-                        Real64 const scriptF(zone_ScriptF[lSR]); // [ lSR ] == ( SendZoneSurfNum+1, RecZoneSurfNum+1 )
+                if (CarrollMethod) {
+                    if (rec_construct.TypeIsWindow) {
+                        Real64 CarrollMRTInKTo4thWin = CarrollMRTInKTo4th;  // arbitrary value, IR will be zero
+                        Real64 CarrollMRTNumeratorWin(0.0);
+                        Real64 CarrollMRTDenominatorWin(0.0);
+                        for (size_type SendZoneSurfNum = 0; SendZoneSurfNum < s_zone_Surfaces; ++SendZoneSurfNum) {
+                            int const SendSurfNum = zone_SurfacePtr[SendZoneSurfNum];
+                            auto const &send_surface_window(SurfaceWindow(SendSurfNum));
+                            int const ConstrNumSend = Surface(SendSurfNum).Construction;
+                            auto const &send_construct(Construct(ConstrNumSend));
+
+                            Real64 SendSurfTemp;
+                            if (SendZoneSurfNum != RecZoneSurfNum) {
+                                if (send_construct.WindowTypeEQL || send_construct.WindowTypeBSDF) {
+                                    SendSurfTemp = send_surface_window.EffInsSurfTemp;
+                                } else if (send_construct.TypeIsWindow && send_surface_window.OriginalClass != SurfaceClass_TDD_Diffuser) {
+                                    if (SurfIterations == 0 && send_surface_window.ShadingFlag <= 0) {
+                                        SendSurfTemp = send_surface_window.ThetaFace(2 * send_construct.TotGlassLayers) - KelvinConv;
+                                    } else if (send_surface_window.ShadingFlag == IntShadeOn || send_surface_window.ShadingFlag == IntBlindOn) {
+                                        SendSurfTemp = send_surface_window.EffInsSurfTemp;
+                                    } else {
+                                        SendSurfTemp = SurfaceTemp(SendSurfNum);
+                                    }
+                                } else {
+                                    SendSurfTemp = SurfaceTemp(SendSurfNum);
+                                }
```

Comment:
These lines appear to be identical to the logic in lines 348:360. Why not save SendSurfTemp in an array and just grab it here instead of going through all these gymnastics all over again? Seems that would improve the CMRT speed.

And then we're doing all the same stuff again for RecSurfs in 394:420? This one gets emissivities too, but seems we could do that up in 348:360, save arrays of temps and emissivities and get rid of all of these ifs.

Reply:
Here is something else to consider. There is a possibility that we will want to run both ScriptF and CarrollMRT within the same simulation. For the same enclosure. In the same timestep. My guess is that the difference between ScriptF and CarrollMRT is not uniform, but rather geometry dependent. I can see an optimization where we run both for all enclosures for a few time steps and then choose the algorithm on an enclosure basis based on whether the two algorithms produce similar results within threshold (CarrollMRT) or not (ScriptF).

Reply:
@mjwitte I like your suggestion. Let me see if I can clean it up a bit.

It's always a question of where to stop. This whole file could probably use a refactor/redesign, especially if we are moving towards enclosures as a concept. We could make an `Enclosure` class that contains many of these methods broken out into a cleaner design to avoid repetition.